Userland/Applications/PixelPaint/Tools/BrushTool.cpp

```diff
@@ -172,4 +177,22 @@ GUI::Widget* BrushTool::get_properties_widget()
     return m_properties_widget.ptr();
 }
 
+NonnullRefPtr<Gfx::Bitmap> BrushTool::build_cursor()
+{
+    // FIXME: changing the image scale does currently not trigger a cursor update :-/
+    //        only changes to the brush size will take account for the current scale setting
```

Comment:
I think it's a neat feature to have the a cursor/preview that shows the bush size, but I'm not sure this is the best way to implement it. 

Rather than generating a bitmap with the circle + crosshair setting that to the cursor, I think (though I could be wrong here, so feel free to correct me), you might be able to keep the current crosshair cursor as-is, and instead draw the brush size circle in the `on_second_paint()` for this tool (similar to how the other previews are done). If that works, it should be fairly easy to get it working with the editor scaling :^)

Reply:
Hey @MacDue,
i will check and try if a implementation with on_second_paint can achive the same features - but for today i didn't had the time for that.

My frist concerns related the on_second_paint:
- the painting seems to be bound to a layer where the layer must not fill the full editor canvas, so that might feel awkward when hovering over different layers
- currently the cursor is already changed while moving the size slider to give an instant response how big the brush would be, not sure if that could be covered with a second piant from the slider component. Would be awsome if that feature could be retained with a different impelemtation. Any implemtation recommendations are welcome :)

Regards,
Torsten

Reply:
No worries, I think you might be right about the layer thing (I'd have to have a proper look). What happens right now if you zoom if while using the paint brush? I assume the cursor appears too small?

Reply:
Hey @MacDue,
just a short update on my second paint attempt.

For the moment i don't have a solution that is working as expected. I was not able to get the mouse cursor drawn when just moving the mouse over the canvas - that was just working when forcing the editor to update itself, that seems somehow wrong either.
If we need to avoid  generating the cursor bitmap by our own, then i think i need some more suggestions on how to achieve that. Or switching to a implementation without generating a cursor bitmap is a topic for a future pr...

Regards